src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+        let receiver = try!(OsIpcReceiver::new_named());
+        let pipe_id = receiver.pipe_id.unwrap().to_string();
+        Ok((
+            OsIpcOneShotServer {
+                receiver: RefCell::new(Some(receiver)),
+            },
+            pipe_id
+        ))
+    }
+
+    pub fn accept(&self) -> Result<(OsIpcReceiver,
+                                    Vec<u8>,
+                                    Vec<OsOpaqueIpcChannel>,
+                                    Vec<OsIpcSharedMemory>),WinError> {
+        let mut receiver = self.receiver.borrow_mut().take().unwrap();
+        try!(receiver.accept());
+        let (data, channels, shmems) = try!(receiver.recv());
+        Ok((receiver, data, channels, shmems))
+    }
+}
+
+pub enum OsIpcChannel {
+    Sender(OsIpcSender),
+    Receiver(OsIpcReceiver),
+}
+
+impl OsIpcChannel {
+}
+
+#[derive(PartialEq, Debug)]
+pub struct OsOpaqueIpcChannel {
+    is_sender: bool,
+    handle: HANDLE,
+}
+
+impl OsOpaqueIpcChannel {
+    fn new(is_sender: bool, handle: HANDLE) -> OsOpaqueIpcChannel {
+        OsOpaqueIpcChannel {
+            is_sender: is_sender,
+            handle: handle,
+        }
+    }
+
+    pub fn to_receiver(&self) -> OsIpcReceiver {
+        assert!(!self.is_sender);
+        OsIpcReceiver::from_handle(self.handle)
+    }
+
+    pub fn to_sender(&self) -> OsIpcSender {
+        assert!(self.is_sender);
+        OsIpcSender::from_handle(self.handle)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct WinError(pub u32);
+
+impl WinError {
+    pub fn error_string(errnum: u32) -> String {
+        // This value is calculated from the macro
+        // MAKELANGID(LANG_SYSTEM_DEFAULT, SUBLANG_SYS_DEFAULT)
+        let lang_id = 0x0800 as winapi::DWORD;
+        let mut buf = [0 as winapi::WCHAR; 2048];
+
+        unsafe {
+            let res = kernel32::FormatMessageW(winapi::FORMAT_MESSAGE_FROM_SYSTEM |
+                                               winapi::FORMAT_MESSAGE_IGNORE_INSERTS,
+                                               ptr::null_mut(),
+                                               errnum as winapi::DWORD,
+                                               lang_id,
+                                               buf.as_mut_ptr(),
+                                               buf.len() as winapi::DWORD,
+                                               ptr::null_mut()) as usize;
+            if res == 0 {
+                // Sometimes FormatMessageW can fail e.g. system doesn't like lang_id,
+                let fm_err = kernel32::GetLastError();
+                return format!("OS Error {} (FormatMessageW() returned error {})",
+                               errnum, fm_err);
+            }
+
+            match String::from_utf16(&buf[..res]) {
+                Ok(mut msg) => {
+                    // Trim trailing CRLF inserted by FormatMessageW
+                    let len = msg.trim_right().len();
+                    msg.truncate(len);
```

Comment:
This is debug only code, but it should be fine regardless; worst case we get an extra newline.  But I like `msg.trim().to_string()` better anyway :)